GettingStarted_4.0.0-rc1.md

```diff
@@ -0,0 +1,115 @@
+
+
+This flag is by default set to zero.
+
+New API's in 4.0.0-rc1:
+----------------------
+Change 1:
+
+1. Old API: FreeRTOS_IPInit
+2. New API: FreeRTOS_IPInit_Multi
+3. Backward Compatibility - Yes
+4. Change:
+    1. FreeRTOS_IPInit - Backward compatibility with the IPv4 FreeRTOS+TCP main branch which only supports single network interface. Can still be used for single interface IPv4 systems.
+    2. FreeRTOS_IPInit_Multi - It supports multiple interfaces. Before calling this function, at least 1 interface and 1 end-point must have been set-up. FreeRTOS_IPInit_Multi() replaces the earlier FreeRTOS_IPInit(). 
+
+Change 2:
+
+1. Old API: FreeRTOS_GetAddressConfiguration/FreeRTOS_SetAddressConfiguration
+2. New API: FreeRTOS_GetEndPointConfiguration/FreeRTOS_SetEndPointConfiguration 
+3. Backward Compatibility - Yes
+4. Change:
+    1. FreeRTOS_GetAddressConfiguration/FreeRTOS_SetAddressConfiguration - used to get/set the address configuration from the global variables initialised during FreeRTOS_IPInit
+    2. FreeRTOS_GetEndPointConfiguration/FreeRTOS_SetEndPointConfiguration will get/set the same address configuration from/to the end point.
+
+Change 3:
+
+1. Old API:  void * FreeRTOS_GetUDPPayloadBuffer( size_t uxRequestedSizeBytes,
+                                                TickType_t uxBlockTimeTicks) 
+2. New API: void * FreeRTOS_GetUDPPayloadBuffer_Multi( size_t uxRequestedSizeBytes,
+                                                     TickType_t uxBlockTimeTicks, uint8_t ucIPType ) 
+3. Backward Compatibility - Yes
+4. Change:
+    1. FreeRTOS_GetUDPPayloadBuffer - Backward compatibility with the IPv4 FreeRTOS+TCP main branch. Can still be used for IPv4 use cases.
+    2. FreeRTOS_GetUDPPayloadBuffer-  Add a new argument (uint8_t ucIPType) to specify IP type to support both IPv4 and IPv6
+
+Change 5:
+
+    1. Old API: pxFillInterfaceDescriptor
+    2. New API: prefix_pxFillInterfaceDescriptor 
+        1. where prefix = Network Interface Name 
+        2. E.g pxWinPcap_FillInterfaceDescriptor
+    3. Backward Compatibility - Yes
+    4. Change:
+        1. pxFillInterfaceDescriptor - It is there for downward compatibility. The function FreeRTOS_IPInit() will call it to initialise the interface and end-point objects
+        2. pxWinPcap_FillInterfaceDescriptor - New function with the same functionality.
+
+Change 6:
+
+    1. Old API: void vApplicationIPNetworkEventHook( eIPCallbackEvent_t eNetworkEvent )
+    2. New API: New argument “NetworkEndPoint_t * pxEndPoint” added
+        1. void vApplicationIPNetworkEventHook( eIPCallbackEvent_t eNetworkEvent, struct xNetworkEndPoint * pxEndPoint )
+    3. Backward Compatibility - Yes
+    4. Change:
+        1. New argument “struct xNetworkInterface * pxNetworkInterface” added.
+        2. We are adding ipconfigIPv4_BACKWARD_COMPATIBLE flag to differentiate between old API and new API.
+
+    Code Snippet:
```

Comment:
Should we move this at the end as an expample snippet?

Reply:
Done